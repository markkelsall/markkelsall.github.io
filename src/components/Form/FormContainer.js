import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {getPickUpLocations} from '../../actions/pickUpLocationsActions';
import {withRouter} from 'react-router-dom';

import ListItem from '../ListItem/ListItem';
import NoResultsItem from '../NoResultsItem/NoResultsItem';

import loaderImage from '../../loader.gif';
import './Form.css'

class FormContainer extends Component {

    static propTypes = {
        getPickUpLocations: PropTypes.func,
        searchResults: PropTypes.object
    }

    constructor () {
        super(...arguments);
        this.state = {
            focus: false,
            searchTerm: '',
            searchResults: {
                items: []
            }
        };
    }

    onFocus = () => {
        if (this.state.searchTerm.length >= 2) {
            this.setState({
                focus: true
            });
        }
    }

    onBlur = () => {
        this.setState({
            focus: false
        });
    }

    handleChange = event => {
        const searchTerm = event.target.value;
        if (event.target.value && event.target.value.length >= 2) {
            this.props.getPickUpLocations(searchTerm);
            this.setState({
                searchTerm,
                focus: true
            });
        } else {
            this.setState({
                searchTerm,
                searchResults: {
                    items: []
                },
                focus: false
            });
        }
    }

    render = () => (
        <div className="formContainer">
            <fieldset id="location-fieldset">
                <div className="pu-fts-container">
                    <label htmlFor="ftsAutocomplete">Pick-up Location</label>
                    {this.props.searching && <img alt="searching..." className="pu-fts-loader" src={loaderImage}/>}
                    <input type="text" onChange={this.handleChange} 
                        onFocus={this.onFocus} onBlur={this.onBlur}
                        id="ftsAutocomplete" name="ftsAutocomplete"
                        autoComplete="off" placeholder="city, airport, station, region, districtâ€¦"
                        aria-required="true" aria-autocomplete="list" aria-haspopup="true"/>
                </div>
                <ul className={`ui-autocomplete ui-corner-all ${this.state.focus ? "showSearchResults" : "hideSearchResults"}`} role="listbox" aria-activedescendant="ui-active-menuitem" tabIndex={0}>
                    {this.props.searchResults.items.length > 0 ?
                        this.props.searchResults.items.map((item, key) => <ListItem item={item} key={key} searchTerm={this.state.searchTerm}/>)
                        : <NoResultsItem/>
                    }
                </ul>
            </fieldset>
        </div>
    )
}

export default withRouter(connect(
    state => ({
        searchResults: state.pickUp.pickupLocationSearchResults,
        searching: state.pickUp.searching
    }),
    { getPickUpLocations }
)(FormContainer))
