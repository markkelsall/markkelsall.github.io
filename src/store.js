import {createStore, applyMiddleware, combineReducers} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import pickupReducer from './reducers/pickUpReducer';

const reducer = combineReducers({
  pickUp: pickupReducer
})

export default createStore(
  reducer,
  composeWithDevTools(
    applyMiddleware(thunk)
  )
)