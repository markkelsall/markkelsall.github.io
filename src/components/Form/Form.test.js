import React from 'react'
import renderer from 'react-test-renderer'
import FormContainer from './FormContainer'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../store'

it ('Should render properly', () => {
    const component = renderer.create(
        <Provider store={store}><Router><FormContainer /></Router></Provider>,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})