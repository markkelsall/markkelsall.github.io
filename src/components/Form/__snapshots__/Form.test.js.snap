// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render properly 1`] = `
<div
  className="formContainer"
>
  <fieldset
    id="location-fieldset"
  >
    <div
      className="pu-fts-container"
    >
      <label
        htmlFor="ftsAutocomplete"
      >
        Pick-up Location
      </label>
      <input
        aria-autocomplete="list"
        aria-haspopup="true"
        aria-required="true"
        autoComplete="off"
        id="ftsAutocomplete"
        name="ftsAutocomplete"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="city, airport, station, region, districtâ€¦"
        type="text"
      />
    </div>
    <ul
      aria-activedescendant="ui-active-menuitem"
      className="ui-autocomplete ui-corner-all hideSearchResults"
      role="listbox"
      tabIndex={0}
    >
      <li
        className="ui-menu-item"
        role="menuitem"
        tabIndex="-1"
      >
        <a
          className="ui-state-disabled"
        >
          <span>
            No results found
          </span>
        </a>
      </li>
    </ul>
  </fieldset>
</div>
`;
